src/Rector/Class_/CommandPropertyToAttributeRector.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Rector\Symfony\Rector\Class_;
+
+use Rector\Core\Rector\AbstractRector;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Rector\PhpAttribute\Printer\PhpAttributeGroupFactory;
+use Rector\Php80\NodeAnalyzer\PhpAttributeAnalyzer;
+use Rector\Core\ValueObject\PhpVersionFeature;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use PHPStan\Type\ObjectType;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Arg;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use function class_exists;
+
+/**
+ * @see https://symfony.com/doc/current/console.html#registering-the-command
+ *
+ * @see \Rector\Symfony\Tests\Rector\Class_\CommandPropertyToAttributeRector\CommandPropertyToAttributeRectorTest
+ */
+final class CommandPropertyToAttributeRector extends AbstractRector implements MinPhpVersionInterface
+{
+    private const ATTRIBUTE = 'Symfony\Component\Console\Attribute\AsCommand';
+
+    public function __construct(
+        /**
+         * @readonly
+         */
+        private readonly PhpAttributeGroupFactory $phpAttributeGroupFactory,
+        /**
+         * @readonly
+         */
+        private readonly PhpAttributeAnalyzer $phpAttributeAnalyzer
+    )
+    {
+    }
+
+    public function provideMinPhpVersion() : int
+    {
+        return PhpVersionFeature::ATTRIBUTES;
+    }
+
+    public function getRuleDefinition() : RuleDefinition
+    {
+        return new RuleDefinition('Add Symfony\Component\Console\Attribute\AsCommand to Symfony Commands and remove the deprecated properties', [new CodeSample(<<<'CODE_SAMPLE'
+class SunshineCommand extends \Symfony\Component\Console\Command\Command
+{
+    /** @var string|null */
+    public static $defaultName = 'sunshine';
+}
+CODE_SAMPLE
+            , <<<'CODE_SAMPLE'
+#[\Symfony\Component\Console\Attribute\AsCommand('sunshine')]
+class SunshineCommand extends \Symfony\Component\Console\Command\Command
+{
+}
+CODE_SAMPLE)]);
+    }
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes() : array
+    {
+        return [Class_::class];
+    }
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node) : ?Node
+    {
+        if (!$this->isObjectType($node, new ObjectType('Symfony\\Component\\Console\\Command\\Command'))) {
+            return null;
+        }
+
+        if (!class_exists(self::ATTRIBUTE)) {
```

Comment:
it should use reflectionProvider instead:

```suggestion
        if (! $this->reflectionProvider->hasClass(self::ATTRIBUTE)) {
```

You can inject dependency to `__construct()`:

```php
private readonly \PHPStan\Reflection\ReflectionProvider $reflectionProvider
```